src/TypeReflectors/PromotedPropertyTypeReflector.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Spatie\TypeScriptTransformer\TypeReflectors;
+
+use ReflectionProperty;
+use ReflectionType;
+
+class PromotedPropertyTypeReflector extends TypeReflector
+{
+    public static function create(ReflectionProperty $reflection): self
+    {
+        return new self($reflection);
+    }
+
+    public function __construct(ReflectionProperty $reflection)
+    {
+        parent::__construct($reflection);
+    }
+
+    protected function getDocblock(): string
+    {
+        return $this->reflection->getDeclaringClass()->getMethod('__construct')->getDocComment();
+    }
+
+    protected function docblockRegex(): string
+    {
+        $parameterName = $this->reflection->getName();
+        return '/@param\s+([^\s<]+(?:<[^>]*>)?)\s+\$' . preg_quote($parameterName, '/') . '\b/m';
+
+    }
+
+    protected function getReflectionType(): ?ReflectionType
+    {
+        return $this->reflection->getType();
+    }
+
+    protected function getAttributes(): array
+    {
+        return $this->reflection->getAttributes();
+    }
+}
```

Comment:
Could we just use the `PropertyTypeReflector` here? Since a promoted property is technically a property. We could in the constructor check if the reflection property is a promoted property and then change `getDocblock` and `docblockRegex`. In v3 this problem should already be fixed since we're using PHPstan annotations over there.